src/t8_cmesh/t8_cmesh_testcases.c

```diff
@@ -310,8 +310,8 @@ t8_test_create_new_hypercube_cmesh (int cmesh_id)
 
   if ((int) eclass == 7) {
```

Comment:
Please remove the int-conversion and replace it with an comparison with T8_ECLASS_PYRA